numba/cpython/builtins.py

```diff
@@ -750,7 +750,8 @@ def false_impl(var, typs):
                         types.DictType, types.LiteralStrKeyDict, types.List,
                         types.ListType, types.Tuple, types.UniTuple, types.Set,
                         types.Function, types.ClassType, types.UnicodeType,
-                        types.ClassInstanceType, types.NoneType, types.Array)
+                        types.ClassInstanceType, types.NoneType, types.Array,
+                        types.UnicodeCharSeq)
```

Comment:
It would be good to add a test for this additional type, perhaps something based on:
```python
from numba import njit
import numpy as np

@njit
def foo(a):
    unichr_inst = a.take(0)
    isinstance(unichr_inst, float)

a = np.array(['a']).take(0)

foo(a)
```
as a new test within https://github.com/numba/numba/blob/77882b19460110d28bcd617e1aca37a2899d0513/numba/tests/test_builtins.py#L1209